keras/backend/theano_backend.py

```diff
@@ -472,7 +472,8 @@ def _old_normalize_batch_in_training(x, gamma, beta,
     '''Computes mean and std for batch then apply batch_normalization on batch.
     '''
     dev = theano.config.device
-    use_cudnn = ndim(x) < 5 and reduction_axes == [0, 2, 3] and (dev.startswith('cuda') or dev.startswith('gpu'))
+    use_cudnn = ndim(x) < 5 and reduction_axes == [0, 2, 3] and (dev.startswith('cuda') or dev.startswith('gpu')) and \
+        theano.sandbox.cuda.dnn_available()
```

Comment:
this is not good I think with the new back-end. I think it should be:

    use_cudnn = ndim(x) < 5 and reduction_axes == [0, 2, 3]
    if dev.startswith('gpu') and theano.sandbox.cuda.dnn_available():
        pass
    elsif dev.startswith('cuda') and theano.gpuarray.dnn.dnn_available(dev):
        pass
    else: use_cudnn = False

This can be refactored, but I think it show more clearly the logic to use.